src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -819,6 +847,78 @@ DWORD CProfilerManager::CalculateEventMask(DWORD dwAdditionalFlags)
     return result;
 }
 
+HRESULT CProfilerManager::InitializeCore(
+    _In_ IUnknown* pCorProfilerInfoUnk,
+    _In_ const vector<CComPtr<CConfigurationSource>>& configSources
+    )
+{
+    IfNullRetPointer(pCorProfilerInfoUnk);
+
+    HRESULT hr = S_OK;
+
+    IfFailRet(pCorProfilerInfoUnk->QueryInterface(__uuidof(ICorProfilerInfo), (LPVOID*)&m_pRealProfilerInfo));
+
+    IfFailRet(DetermineClrVersion());
+
+    //set event mask to be default one. Later host and Instrumentation methods may call it again
+    IfFailRet(this->SetEventMask(m_dwEventMask));
```

Comment:
>SetEventMask [](start = 20, length = 12)

Are there limitations on SetEventMask for attach vs. startup?

Reply:
Yep, [https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/profiling/davbr-blog-archive/Attach.md#specific-callback-limitations](https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/profiling/davbr-blog-archive/Attach.md#specific-callback-limitations)

This PR won't address these issues, but future ones would.

---
In reply to: [381591393](https://github.com/microsoft/CLRInstrumentationEngine/pull/204#discussion_r381591393) [](ancestors = 381591393)

Reply:
#205